packages/core/src/transports/blockchair/getUTXOs.ts

```diff
@@ -0,0 +1,126 @@
+import type { UTXO } from '../../types/transaction.js'
+import { urlWithParams } from '../../utils/url.js'
+import type { RpcMethodHandler } from '../types.js'
+import type {
+  BlockChairDashboardAddressResponse,
+  BlockchairAddressBalanceData,
+  BlockchairResponse,
+  BlockchairUTXO,
+} from './blockchair.types.js'
+
+const blockChairUTXOTransformer =
+  (scriptHex: string) =>
+  (data: BlockchairUTXO): UTXO => ({
+    blockHeight: data.block_id,
+    scriptHex,
+    txId: data.transaction_hash,
+    value: data.value,
+    vout: data.index,
+  })
+
+export const getUTXOs: RpcMethodHandler<'getUTXOs'> = async (
+  client,
+  { baseUrl, apiKey },
+  { address, minValue }
+) => {
+  async function isAddressUTXOEnough(minValue: number): Promise<boolean> {
+    const apiUrl = urlWithParams(`${baseUrl}/addresses/balances`, {
+      key: apiKey,
+      addresses: address,
+    })
+    const response = (await client.request({
+      url: apiUrl,
+      fetchOptions: { method: 'GET' },
+    })) as unknown as BlockchairResponse<BlockchairAddressBalanceData>
+
+    if (response.context?.code !== 200) {
+      throw new Error(response.context.error || 'Error fetching user balance')
+    }
+
+    const addressUTXOBalance = Number(response.data[address])
+
+    return addressUTXOBalance > minValue
+  }
+
+  async function* fetchUTXOs() {
+    let offset = 0
+    let hasMore = true
+
+    while (hasMore) {
+      const apiUrl = urlWithParams(
+        `${baseUrl}/dashboards/addresses/${address}`,
+        {
+          key: apiKey,
+          offset: `0,${offset}`,
+        }
+      )
+      const response = (await client.request({
+        url: apiUrl,
+        fetchOptions: { method: 'GET' },
+      })) as unknown as BlockchairResponse<BlockChairDashboardAddressResponse>
+
+      // blockchair returns a 404 for empty addresses
+      if (response.context?.code !== 200 && response.context?.code !== 404) {
+        throw new Error(`${response.context.code} : ${response.context?.error}`)
```

Comment:
Similarly, here, we should make errors more consistent and handleable 😁

Reply:
If we don't have UTXOs for a specific address, that should be a balance error similar to what we will return when minValue is unmet.

Reply:
In this case, this code would only be reached if minValue is not specified, and the address has no utxo.
This is not an error, but the API returns a 404, so we need to add 404 error to our whitelist. 

Therefore, we only throw an error if the code is not 200 and not 404, makes sense?